apps/api/bronya.config.ts

```diff
@@ -97,12 +107,41 @@ export const esbuildOptions: BuildOptions = {
    * @RFC What would be the best way to resolve these two values?
    */
   outExtension: { ".js": ".mjs" },
+
+  plugins: [
+    {
+      name: "in-memory-cache",
+      setup: (build) => {
+        build.onResolve({ filter: /virtual:courses/ }, (args) => ({
+          path: args.path,
+          namespace,
+        }));
+        build.onResolve({ filter: /virtual:instructors/ }, (args) => ({
+          path: args.path,
+          namespace,
+        }));
+        build.onLoad({ filter: /virtual:courses/, namespace }, async () => ({
+          contents: `export const courses = ${JSON.stringify(
+            Object.fromEntries(
+              (await prisma.course.findMany()).map(normalizeCourse).map((x) => [x.id, x]),
+            ),
+          )}`,
+        }));
```

Comment:
Not that anybody would ever look at this devops code or care what it does, but I'd personally implement it something like this so it's easier for me to understand. Same idea applies to the virtual instructors file, if you think it matters 🤓 

```suggestion
        build.onLoad({ filter: /virtual:courses/, namespace }, async () => {
         const allCourses =  await prisma.course.findMany()
         
         const courses = courses
           .map(normalizeCourse)
           .reduce((course) => ({
             [course.id]: course
           }), {} as Record<string, Course>)

         return {
            contents: `export const courses = ${JSON.stringify(courses)}`,
          }
        });
```

Reply:
The reduction into an object doesn't seem to actually work. Plus I think having the callback just return and not have a body is nice, but that's just me.